src/Education/XamlGenerator/IRuleHelpXamlBuilder.cs

```diff
@@ -170,14 +170,33 @@ private void WriteSubTitle(IRuleInfo ruleInfo)
 
         private void WriteSubTitleElement_IssueType(IRuleInfo ruleInfo)
         {
-            // TODO: icon
-            WriteSubTitleElement("Issue type: " + ruleInfo.IssueType.ToString());
+            var imageInfo = SubTitleImageInfo.IssueTypeImages[ruleInfo.IssueType];
+            WriteSubTitleElementWithImage(imageInfo);
         }
 
         private void WriteSubTitleElement_Severity(IRuleInfo ruleInfo)
         {
-            // TODO: icon
-            WriteSubTitleElement("Severity: " + ruleInfo.DefaultSeverity.ToString());
+            var imageInfo = SubTitleImageInfo.SeverityImages[ruleInfo.DefaultSeverity];
+            WriteSubTitleElementWithImage(imageInfo);
+        }
+
+        private void WriteSubTitleElementWithImage(SubTitleImageInfo imageInfo)
+        {
+            writer.WriteStartElement("Span");
+            ApplyStyleToStartElement(StyleResourceNames.SubtitleElement_Span);
+
+            if (imageInfo.ImageResourcePath != null)
+            {
+                writer.WriteStartElement("InlineUIContainer");
+                writer.WriteStartElement("Image");
```

Comment:
`FlowDocument` can only contain elements deriving from `TextElement` e.g. `Paragraph`, `Section`, `Inline`.

`Image` is a "normal" XAML UI control that inherits from `UIElement`, which means it can't be hosted directly in the document. So, we have to add an `InlineUIContainer` first, which allows us to host non-`TextElement` XAML classes in documents.

There is also a `BlockUIContainer` class that we could use to host "normal" XAML classes at `Block` level if we needed to.